tests/unit_tests/udfs/test_base_udf.py

```diff
@@ -14,6 +14,30 @@
 import re
 
 
+class regex_matcher: #this one feels more correct
+    """Assert that a given string meets some expectations."""
+    def __init__(self, pattern, flags=0):
+        self._regex = re.compile(pattern, flags)
+
+    def is_in(self, actual):
+        return bool(self._regex.search(actual))
+
+    def __repr__(self):
+        return self._regex.pattern
+
+
+class regex_matcher2: #this one looks nicer when used and gives better error messages
+    """Assert that a given string meets some expectations."""
+    def __init__(self, some_sting):
+        self._string = some_sting
+
+    def __contains__(self, pattern):
```

Comment:
The __eq__ in the linked example has a reason.  This gives you a good diff. You used it then

`assert actual_value == regex_matcher("abc.*")`